src/server/api/controllers/candidates.controller.js

```diff
@@ -35,11 +36,25 @@ const createCandidate = async (userId, boardId, newCandidate) => {
   if (typeof newCandidate.name !== 'string') {
     throw new IncorrectEntryError(`Candidate name should be string`);
   }
-  const createNewCandidate = await knex('candidates').insert({
+  await checkUserRole({ userId, boardId });
+  const createNewCandidate = await knex('candidates').returning('id').insert({
     boardId,
     name: newCandidate.name,
     isBlocked: false,
   });
+  const getAllMembersOfBoard = await getMembers.getAllMembers(userId, boardId);
+  const allMembersId = getAllMembersOfBoard.map((member) => member.userId);
+  allMembersId.forEach((id) => {
+    const newBallot = async () => {
+      await knex('ballots').insert({
+        boardId,
+        userId: id,
+        candidateId: createNewCandidate[0],
+        rank: -1,
+      });
+    };
+    newBallot();
+  });
```

Comment:
The idea is certainly correct here, loop over the ids and insert the candidate for each of them. However, you define a new async function inside of the anonymous function of the loop. And then you call that function without awaiting it. This doesn't look right to me. Maybe @senner007 knows a good solution. One approach that I found online is to map the results and await all of the promises. Something like this:

```suggestion
  await Promise.all(allMembersId.map(async (id) => {
      await knex('ballots').insert({
        boardId,
        userId: id,
        candidateId: createNewCandidate[0],
        rank: -1,
      });
  }));
```

Reply:
For clarity you might also store the id and use that in the loop instead.

```js
const newCandidateId = createNewCandidate[0];
```

Reply:
Almost
the map() function does not take asyncronous callbacks.
So like this instead:

```
  await Promise.all(allMembersId.map((id) => {
      return knex('ballots').insert({
        boardId,
        userId: id,
        candidateId: createNewCandidate[0],
        rank: -1,
      });
  }));
```

Reply:
Ahh, sorry it does take async callbacks. But no reason to use it here.